### PR TITLE
Add further DisplayConfig settings

### PR DESCRIPTION
**Pull request by saschagrunert**
Hey there, please take a look at my proposed changes.

- Added the following DisplayConfig settings:
    -  always_on_top
    -  decorations
    -  maximized
    -  multitouch
    -  resizable
    -  transparent
- Update the RenderBuilder to take the DisplayConfig and WindowBuilder
  conversions into account.
- Remove duplicate conversions
- Fix build for amethyst_renderer

### CODE REVIEW

**Comment by saschagrunert** [comment]
Ah I see that my changes are also within #1375 #1379, which was absolutely not my intention. Sorry for that. :innocent: Do we want to merge that first or should be simply use my PR?

File: amethyst_renderer/src/config.rs [hunk 1]
```diff
@@ -17,88 +17,114 @@ use winit::{self, dpi::LogicalSize, Icon, WindowBuilder};
...
 ///     multisampling: 0,
 ///     visibility: true,
-///     vsync: true
+///     vsync: true,
+///     always_on_top: false,
+///     decorations: true,
+///     maximized: false,
+///     multitouch: true,
+///     resizable: true,
+///     transparent: false
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply amethyst_renderer/src/config.rs]: I just followed the existing approach in not adding a trailing comma. So should I add one?

[the remark this answers is not in the document]
Comment (azriel91) [reply amethyst_renderer/src/config.rs]: Yeap, it'll be consistent with what rustfmt does

**Comment by azriel91** [comment]
@khionu good to merge?

**Comment by khionu** [comment]
Bors can't squash. Needs to be squashed manually.

**Comment by saschagrunert** [comment]
I squashed the commits manually.

**Comment by azriel91** [comment]
Rust beta, one timed out, one test went 2% too slow `._.`

bors retry

**Comment by azriel91** [comment]
bors retry

**Comment by azriel91** [comment]
bors retry

**Comment by azriel91** [comment]
bors retry

**Comment by jojolepro** [comment]
If it fails again, merge manually @azriel91

**Comment by saschagrunert** [comment]
Awesome, thanks!

**Pull request merged**